### PR TITLE
Fix user and conference removal

### PR DESCRIPTION
**Pull request by HedicGuibert**
Refs https://github.com/jolicode/starfleet/issues/93

Set `null` on related entities when removing a user or a conference

### CODE REVIEW

File: src/Controller/Admin/AdminController.php [hunk 1]
```diff
@@ -225,4 +233,70 @@ protected function persistTalkEntity(Talk $talk, Form $newForm)
...
+        }
+
+        $participations = $this->participationRepository->findBy(['conference' => $conference]);
+
+        foreach ($participations as $participation) {
+            $participation->setConference(null);
+        }
+
+        $this->em->remove($conference);
+        $this->em->flush();
```
Comment (welcoMattic) [inline src/Controller/Admin/AdminController.php:300]: You must centralize the duplicated code in a private method then call it in the 3 remove*ConferencesEntity methods

**Review by welcoMattic — CHANGES_REQUESTED** [review]

**Review by welcoMattic — APPROVED** [review]

File: src/Controller/Admin/AdminController.php [hunk 2]
```diff
@@ -26,13 +30,17 @@
 
 class AdminController extends EasyAdminController
 {
-    protected $workflowRegistry;
-    protected $logger;
+    protected WorkflowRegistry $workflowRegistry;
+    protected ParticipationRepository $participationRepository;
+    protected SubmitRepository $submitRepository;
+    protected ?LoggerInterface $logger;
```
Comment (lyrixx) [inline src/Controller/Admin/AdminController.php:36]: There is always a logger defined

File: src/Controller/Admin/AdminController.php [hunk 3]
```diff
@@ -225,4 +233,49 @@ protected function persistTalkEntity(Talk $talk, Form $newForm)
...
+        }
+
+        $participations = $this->participationRepository->findBy(['conference' => $conference]);
+
+        foreach ($participations as $participation) {
+            $participation->setConference(null);
+        }
+
+        $this->em->remove($conference);
+        $this->em->flush();
```
Comment (lyrixx) [inline src/Controller/Admin/AdminController.php:279]: If you configure property the DB (and doctrine), The DB will do it for you.
Here, you load tons of data ; you should not. The DB must handle that

File: migrations/Version20210208110825.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * For the full copyright and license information,
+ * please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace DoctrineMigrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+final class Version20210208110825 extends AbstractMigration
```
Comment (lyrixx) [inline migrations/Version20210208110825.php:19]: Let's merge this migration with the previous one

File: src/Controller/Admin/AdminController.php [hunk 5]
```diff
@@ -26,13 +29,17 @@
...
+    protected LoggerInterface $logger;
 
-    public function __construct(WorkflowRegistry $workflowRegistry, LoggerInterface $logger)
+    public function __construct(WorkflowRegistry $workflowRegistry, ParticipationRepository $participationRepository, SubmitRepository $submitRepository, ?LoggerInterface $logger = null)
     {
         $this->workflowRegistry = $workflowRegistry;
-        $this->logger = $logger;
+        $this->participationRepository = $participationRepository;
+        $this->submitRepository = $submitRepository;
+        $this->logger = $logger ?: new NullLogger();
```
Comment (lyrixx) [inline src/Controller/Admin/AdminController.php:42]: Looks like this can be revert

File: src/Entity/Participation.php [hunk 6]
```diff
@@ -27,15 +27,15 @@ class Participation
 
     /**
      * @ORM\ManyToOne(targetEntity="App\Entity\Conference", inversedBy="participations", cascade={"persist"})
-     * @ORM\JoinColumn(nullable=false)
+     * @ORM\JoinColumn(nullable=true)
      */
-    private $conference;
+    private ?Conference $conference;
```
Comment (lyrixx) [inline src/Entity/Participation.php:32]: It does not make sens to me. Why would we want to remove a Conference where someone attended?

**Review by lyrixx — COMMENTED** [review]
Acutally, I don't understand the goal of this PR.

Why do we want to support conference removal?
And user removal?

**Comment by HedicGuibert** [comment]
This addresses https://github.com/jolicode/starfleet/issues/93 where we decided to set null when a user or a conference is deleted

**Comment by HedicGuibert** [comment]
So as mentioned in https://github.com/jolicode/starfleet/issues/93 I ended up reverting everything and only disabling the possibility to delete a user or a conference

**Review by lyrixx — APPROVED** [review]
Thanks

**Pull request merged**